### PR TITLE
Alpha

### PR DESCRIPTION
**Pull request by chromaticWaster**
### Description:

closes https://github.com/capitec/omni-components/issues/68

- Date picker component
- Calendar component 
- Keyboard component - in experimental phase

### Screenshots (if appropriate):

**Test results**

image

**Date Picker**

Date picker component

**Calendar** 

Calendar component

### All Submissions:

* [x] I have followed the guidelines in the Contributing document.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

**Review by stefan505 — CHANGES_REQUESTED** [review]

File: src/calendar/Calendar.stories.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+export const Value: ComponentStoryFormat<Args> = {
+    render: (args: Args) => html`
+    <omni-calendar
+        data-testid="test-calendar"
+        .value="${args.value}"
+        >
+    </omni-calendar>
+    `,
+    name: 'Value',
+    description: 'Set the value of the Calendar component, this has to be a valid date in ISOformat',
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/calendar/Calendar.stories.ts]: Changes implemented as requested.

File: src/date-picker/DatePicker.stories.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import { LabelStory, BaseArgs, HintStory, ErrorStory, PrefixStory, SuffixStory, DisabledStory } from '../core/OmniInputStories.js';
+import { ifNotEmpty } from '../utils/Directives.js';
+import expect from '../utils/ExpectDOM.js';
+import { assignToSlot, ComponentStoryFormat, CSFIdentifier, querySelectorAsync, raw } from '../utils/StoryUtils.js';
+import { DatePicker } from './DatePicker';
+
+import './DatePicker.js';
+
+export default {
+    title: 'UI Components/Select',
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/date-picker/DatePicker.stories.ts]: Changes implemented as requested.

File: src/date-picker/DatePicker.stories.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    component: 'omni-select'
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/date-picker/DatePicker.stories.ts]: Changes implemented as requested.

File: src/date-picker/DatePicker.stories.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    <omni-date-picker
+        data-testid="test-date-picker"
+        label="${ifNotEmpty(args.label)}"
+        .value="${args.value}"
+    >
+
+    </omni-date-picker>
+    `,
+    name: 'Value',
+    description: 'Set the current value of the Date picker component.',
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/date-picker/DatePicker.stories.ts]: Changes implemented as requested.

File: src/date-picker/DatePicker.stories.ts [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    <omni-date-picker
+        data-testid="test-date-picker"
+        label="${ifNotEmpty(args.label)}"
+        .value="${args.value}"
+        locale="${args.locale}"
+    >
+    </omni-date-picker>
+    `,
+    name: 'Locale',
+    description: 'Set the current locale of the Date picker component.',
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/date-picker/DatePicker.stories.ts]: Changes implemented as requested.

File: src/date-picker/DatePicker.ts [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+import { live } from 'lit/directives/live.js';
+import { DateTime } from 'luxon';
+import { OmniFormElement } from '../core/OmniFormElement.js';
+import '../calendar/Calendar.js';
+import '../icons/Calendar.icon.js';
+import '../icons/ChevronLeft.icon.js';
+import '../icons/ChevronRight.icon.js';
+
+/**
+ * Control that allows you to select a specific date
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/date-picker/DatePicker.ts]: Changes implemented as requested.

File: src/date-picker/DatePicker.ts [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+        const distanceFromBottom = visualViewport!.height - this.getBoundingClientRect().bottom;
+        if (distanceFromBottom < 270) {
+            this._bottomOfViewport = true;
+        } else {
+            this._bottomOfViewport = false;
+        }
+    }
+
+    // Check the width of the screen to set the internal mobile boolean to true of false.
+    _mobileCheck() {
```
[the remark this answers is not in the document]
Comment (chromaticWaster) [reply src/date-picker/DatePicker.ts]: Function removed as it wont be required.

**Review by stefan505 — APPROVED** [review]

**Pull request merged**